api/gateway-operator/v1alpha1/watchnamespacegrant_types.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&WatchNamespaceGrant{}, &WatchNamespaceGrantList{})
+}
+
+// WatchNamespaceGrant is a grant that allows a trusted namespace to watch
+// resources in the namespace this grant exists in.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +apireference:kgo:include
+// +kong:channels=gateway-operator
+type WatchNamespaceGrant struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the WatchNamespaceGrant.
+	Spec WatchNamespaceGrantSpec `json:"spec,omitempty"`
+
+	// Status is not specified for WatchNamespaceGrant but it may be added in the future.
+}
+
+// WatchNamespaceGrantSpec defines the desired state of an WatchNamespaceGrant.
+//
+// +apireference:kgo:include
+type WatchNamespaceGrantSpec struct {
+	// From describes the trusted namespaces and kinds that can reference the
+	// namespace this grant exists in.
+	//
+	// Support: Core
```

Comment:
Looks like copied from `ReferenceGrant` in the gateway APIs? I think this is not meaningful here.
```suggestion
```

Reply:
Applied in https://github.com/Kong/kubernetes-configuration/pull/403/commits/006bdb1879a20562e5adbe44d9bbd6abc6d8b03c